### PR TITLE
Recursively test children in consistency lint

### LINKED ISSUE #5765: Consistency linter: needs to recursively check parent
I'm not sure this is a good idea, but for discussion: I was reviewing https://github.com/mdn/browser-compat-data/pull/5757, and it lists a subfeature of `onAuthRequired` as supported from Firefox 46, when `onAuthRequired` itself is only supported from Firefox 54.

This seems like a thing we might be able to lint for?

### PR DESCRIPTION
**Pull request by queengooborg**
This PR fixes #5765 by testing the grandchildren of a feature if a child doesn't have its own compatibility data.  All data has now been fixed, so this lint passes and is ready to merge.

### CODE REVIEW

File: test/linter/test-consistency.js [hunk 1]
```diff
@@ -73,17 +74,41 @@ class ConsistencyChecker {
     }
 
     // Check sub-features.
-    const keys = Object.keys(data).filter((key) => key != '__compat');
-    keys.forEach((key) => {
+    this.getSubfeatures(data).forEach((key) => {
```
Comment (caugner) [inline test/linter/test-consistency.js:77]: FWIW Unrelated code smell: The `ConsistencyError` type (defined on lines 22-25) has an extraneous property `feature`, which is not added by `checkSubfeatures()` doesn't add. Given that `feature` (line 64) is directly inferred from `path`, that property might not be used anywhere.

**Review by caugner — APPROVED** [review]
LGTM and can be merged as is, but I have two remarks.

File: test/linter/test-consistency.js [hunk 2]
```diff
@@ -305,9 +328,9 @@ class ConsistencyChecker {
   getVersionAdded(supportStatement) {
     // A convenience function to squash non-real values and previews into null
     const resolveVersionAddedValue = (statement) =>
-      [true, false, 'preview', null].includes(statement.version_added)
+      [true, false, 'preview', null].includes(statement?.version_added)
         ? null
-        : statement.version_added;
+        : statement?.version_added;
```
[the remark this answers is not in the document]
Comment (queengooborg) [reply test/linter/test-consistency.js:331]: The need for ?. was found when going through webextensions and finding that some features didn't have support statements for a specific browser.  If the browser isn't defined, the value should be treated as `undefined`.
